flake.nix

```diff
@@ -1,187 +1,46 @@
+          stat = builtins.readDir dir;
+          f = acc: name:
+            if stat.${name} == "directory" then
+              acc // recurse (segments ++ [ name ])
+            else if name == "default.nix" && segments != [ ] then
+              acc // {
+                "${builtins.concatStringsSep "-" segments}" = dir;
+              }
+            else
+              acc;
+        in
+        builtins.foldl' f { } (builtins.attrNames stat);
 
-      common-cpu-amd = import ./common/cpu/amd;
-      common-cpu-amd-pstate = import ./common/cpu/amd/pstate.nix;
-      common-cpu-intel = import ./common/cpu/intel;
-      common-cpu-intel-cpu-only = import ./common/cpu/intel/cpu-only.nix;
-      common-cpu-intel-kaby-lake = import ./common/cpu/intel/kaby-lake;
-      common-cpu-intel-sandy-bridge = import ./common/cpu/intel/sandy-bridge;
-      common-gpu-amd = import ./common/gpu/amd;
-      common-gpu-amd-sea-islands = import ./common/gpu/amd/sea-islands;
-      common-gpu-amd-southern-islands = import ./common/gpu/amd/southern-islands;
-      common-gpu-intel = import ./common/gpu/intel.nix;
-      common-gpu-nvidia = import ./common/gpu/nvidia/prime.nix;
-      common-gpu-nvidia-nonprime = import ./common/gpu/nvidia;
-      common-gpu-nvidia-disable = import ./common/gpu/nvidia/disable.nix;
-      common-pc = import ./common/pc;
-      common-pc-hdd = import ./common/pc/hdd;
-      common-pc-laptop = import ./common/pc/laptop;
-      common-pc-laptop-acpi_call = import ./common/pc/laptop/acpi_call.nix;
-      common-pc-laptop-hdd = import ./common/pc/laptop/hdd;
-      common-pc-laptop-ssd = import ./common/pc/ssd;
-      common-pc-ssd = import ./common/pc/ssd;
+      modules = recurse [ ];
+    in
+    {
+      nixosModules = modules // {
+        # Backwards compatibility
+        asus-battery = import ./asus/battery.nix;
+        common-cpu-amd-pstate = import ./common/cpu/amd/pstate.nix;
+        common-cpu-intel-cpu-only = import ./common/cpu/intel/cpu-only.nix;
+        common-gpu-intel = import ./common/gpu/intel.nix;
+        common-gpu-nvidia = import ./common/gpu/nvidia/prime.nix;
+        common-gpu-nvidia-disable = import ./common/gpu/nvidia/disable.nix;
+        common-gpu-nvidia-nonprime = import ./common/gpu/nvidia;
+        common-pc-laptop-acpi_call = import ./common/pc/laptop/acpi_call.nix;
+        common-pc-laptop-ssd = import ./common/pc/ssd;
+        lenovo-legion-y530-15ich = import ./lenovo/legion/15ich;
+        lenovo-thinkpad-p1-gen3 = import ./lenovo/thinkpad/p1/3th-gen;
+        lenovo-thinkpad-z13 = import ./lenovo/thinkpad/z/z13;
+        letsnote-cf-lx4 = import ./panasonic/letsnote/cf-lx4;
+        microsoft-surface-go = import ./microsoft/surface/surface-go;
+        microsoft-surface-laptop-amd = import ./microsoft/surface/surface-laptop-amd;
+        microsoft-surface-pro-intel = import ./microsoft/surface/surface-pro-intel;
+        purism-librem-15v3 = import ./purism/librem/15v3;
```

Comment:
This was not auto-generated because `15v3` is a link to `13v3`, there's probably a way to fix that